numpy/lib/function_base.py

```diff
@@ -764,6 +764,19 @@ def copy(a, order='K', subok=False):
     >>> x[0] == z[0]
     False
 
+    Note that, NumPy will not copy ragged object arrays:
```

Comment:
More accurately would be simply that
```suggestion
    Note that, NumPy will not copy the elements of object arrays:
```

You could see the behavior just as clearly with
```python
a = np.array([{}])
b = a.copy()
b[0]['value'] = 1
```

I don't think we should use "ragged arrays" as the example here, but a throwaway remark like "(including arrays within object arrays)" would be fine